Slack.Automation/Promact.Core.Repository/MailSettingRepository/MailSettingRepository.cs

```diff
@@ -0,0 +1,179 @@
+using Promact.Erp.DomainModel.DataRepository;
+using Promact.Erp.DomainModel.Models;
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Promact.Core.Repository.MailSettingRepository
+{
+    public class MailSettingRepository : IMailSettingRepository
+    {
+        #region Private Variables
+        private readonly IRepository<MailSetting> _mailSettingDataRepository;
+        private readonly IOauthCallHttpContextRespository _oauthCallRepository;
+        private readonly IRepository<MailSettingMapping> _mailSettingMappingDataRepository;
+        private readonly IRepository<Group> _groupDataRepository;
+        private IMapper _mapper;
+        #endregion
+
+        #region Constructor
+        public MailSettingRepository(IRepository<MailSetting> mailSettingDataRepository, IOauthCallHttpContextRespository oauthCallRepository,
+            IRepository<MailSettingMapping> mailSettingMappingDataRepository, IRepository<Group> groupDataRepository,
+            IMapper mapper)
+        {
+            _mailSettingDataRepository = mailSettingDataRepository;
+            _oauthCallRepository = oauthCallRepository;
+            _mailSettingMappingDataRepository = mailSettingMappingDataRepository;
+            _groupDataRepository = groupDataRepository;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Public Methods
+        /// <summary>
+        /// Method to get list of project from oauth server
+        /// </summary>
+        /// <returns>list of project</returns>
+        public async Task<List<ProjectAc>> GetAllProjectAsync()
+        {
+            var projects = (await _oauthCallRepository.GetAllProjectsAsync()).FindAll(x=>x.IsActive == true);
+            return (projects);
+        }
+
+        /// <summary>
+        /// Method to get mail setting configuration from projectId
+        /// </summary>
+        /// <param name="projectId">project Id</param>
+        /// <param name="module">name of module{leave, task or scrum}</param>
+        /// <returns>mail setting configuration details</returns>
+        public async Task<MailSettingAC> GetMailSettingDetailsByProjectIdAsync(int projectId, string module)
+        {
+            MailSettingAC mailSetting = new MailSettingAC();
+            var mailSettingDetails = await _mailSettingDataRepository.FirstOrDefaultAsync(x => x.ProjectId == projectId && x.Module == module);
+            if (mailSettingDetails != null)
+            {
+                List<string> listOfTo = new List<string>();
+                List<string> listOfCC = new List<string>();
+                mailSetting = _mapper.Map<MailSetting, MailSettingAC>(mailSettingDetails);
+                var mailSettingMappingList = (await _mailSettingMappingDataRepository.FetchAsync(x => x.MailSettingId == mailSettingDetails.Id)).ToList();
+                var To = mailSettingMappingList.FindAll(x => x.IsTo == true);
```

Comment:
code must be optimized (Not required compare with true, it directly check for true condition)